### PR TITLE
Generic solvers

### PR DESCRIPTION
**Pull request by AminosDz**
Refactor metaheuristics to the Abstract Problem layer (1st layer).

### CODE REVIEW

**Comment by AminosDz** [comment]
- Use inheritance from core.solvers to define family specific parameters

`class lahc(BaseLAHC):

        ls_procedure = FS.FS_LocalSearch()
        generate_neighbour = FS.NeighbourhoodGeneration.deconstruct_construct`

- Rename solvers to BaseSolvers, make lahc, sa ... abstract to prevent from using them directly to solve
- Create a folder named solvers to contain all methods in PM_methods, SM_methods ...

**Comment by AminosDz** [comment]
TODO: Refactor FmriSijkCmax_Instance to FmriSijkCmax and follow the same pattern for the rest.

**Comment by AminosDz** [comment]
@hennimohammed @Neburam 

I just finished refactoring all solvers in pyscheduling using the BaseSolver interface and defined generic solvers, this pull request can (finally) be merged. Let me know if you have feedback about it.

File: src/pyscheduling/FS/FmCmax.py [hunk 1]
```diff
@@ -3,31 +3,25 @@
 from typing import ClassVar, List
 
 import pyscheduling.FS.FlowShop as FlowShop
-import pyscheduling.FS.FS_methods as FS_methods
+from pyscheduling.FS.solvers import BIBA
 import pyscheduling.Problem as Problem
 from pyscheduling.FS.FlowShop import Constraints
 from pyscheduling.Problem import Job, Objective
+from pyscheduling.core.base_solvers.base_solver import BaseSolver
```
Comment (hennimohammed) [inline src/pyscheduling/FS/FmCmax.py:10]: Can be shortened from `import bases_solvers.base_solver.BaseSolver` to `import base_solvers.BaseSolver`

File: src/pyscheduling/FS/FmriSijkCmax.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
```
Comment (hennimohammed) [inline src/pyscheduling/FS/FmriSijkCmax.py:1]: I don't see where the new `field` import is used

**Pull request merged**